pkg/module_manager/module_loader.go

```diff
@@ -0,0 +1,197 @@
+package module_manager
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/flant/addon-operator/pkg/app"
+	"github.com/flant/addon-operator/pkg/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+func SearchModules(modulesDirs string) (*ModuleSet, error) {
+	paths := splitToPaths(modulesDirs)
+	modules := new(ModuleSet)
+	for _, path := range paths {
+		modulesInDir, err := findModulesInDir(path)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add only "new" modules. Modules from first directories are in top priority as commands in $PATH.
+		for _, module := range modulesInDir {
+			if !modules.Has(module.Name) {
+				modules.Add(module)
+			}
+		}
+	}
+
+	return modules, nil
+}
+
+const PathsSeparator = ":"
```

Comment:
So, PathSeletor is a dedicated single-use constant. While "values.yaml" is just a magic string!

Reply:
Good point, I'll extract it to the constant.